core-plugins/src/test/java/co/cask/hydrator/plugin/batch/action/OracleExportActionTest.java

```diff
@@ -0,0 +1,170 @@
+    // Shutdown MiniDFSCluster
+    dfsCluster.shutdown();
+    folder.delete();
+  }
+
+  @Before
+  public void setup() throws Exception {
+    sshServer = setupSSHDaemon();
+    sshServer.start();
+  }
+
+  @After
+  public void cleanup() throws Exception {
+    if (sshServer != null) {
+      sshServer.stop();
+    }
+  }
+
+  private SshServer setupSSHDaemon() throws Exception {
+    SshServer sshServer = SshServer.setUpDefaultServer();
+    SimpleGeneratorHostKeyProvider simpleGeneratorHostKeyProvider = new SimpleGeneratorHostKeyProvider();
+    simpleGeneratorHostKeyProvider.setAlgorithm(KeyUtils.RSA_ALGORITHM);
+    sshServer.setKeyPairProvider(simpleGeneratorHostKeyProvider);
+    sshServer.setPasswordAuthenticator(new PasswordAuthenticator() {
+      @Override
+      public boolean authenticate(String username, String password, ServerSession session) {
+        return username.equals(USER) && password.equals(PASSPHRASE);
+      }
+    });
+    sshServer.setShellFactory(new ProcessShellFactory(SHELL_ARGS));
+    sshServer.setCommandFactory(EchoCommandFactory.INSTANCE);
+    return sshServer;
+  }
+
+  @Test
+  public void testCheckWrongArgumentForServerPassword() throws Exception {
+    OracleExportAction.OracleExportActionConfig oracleExportActionConfig =
+      new OracleExportAction.OracleExportActionConfig(sshServer.getHost(),
+                                                      sshServer.getPort(), USER, "wrongPassword", "system",
+                                                      "cask", "home", "sid", "/tmp/junit.csv",
+                                                      "select * from test;", "csv");
+    OracleExportAction oracleExportAction = new OracleExportAction(oracleExportActionConfig);
+    MockActionContext mockActionContext = new MockActionContext();
+    try {
+      oracleExportAction.run(mockActionContext);
+    } catch (IOException e) {
+      Assert.assertEquals("SSH authentication error when connecting to test@" + sshServer.getHost() + " on port " +
+                            sshServer.getPort(), e.getMessage());
+    }
+  }
+
+  @Test
+  public void testCheckWrongArgumentForQuery() throws Exception {
+    OracleExportAction.OracleExportActionConfig oracleExportActionConfig =
+      new OracleExportAction.OracleExportActionConfig(sshServer.getHost(),
+                                                      sshServer.getPort(), USER, PASSPHRASE, "system",
+                                                      "cask", "home", "sid", "/tmp/junit.csv",
+                                                      "select * from test", "csv");
+    OracleExportAction oracleExportAction = new OracleExportAction(oracleExportActionConfig);
+    MockActionContext mockActionContext = new MockActionContext();
+    try {
+      oracleExportAction.run(mockActionContext);
+    } catch (IllegalArgumentException e) {
+      Assert.assertEquals("Query should have ; at the end", e.getMessage());
```

Comment:
hmmm, this test should not be needed now right, because you handle the scenario where the ``;`` is not present? If this test passes, it seems like there's something wrong with the code.